sql/src/main/java/io/crate/planner/projection/FetchProjection.java

```diff
@@ -55,14 +58,60 @@ public FetchProjection(int collectPhaseId,
                            TreeMap<Integer, String> readerIndices,
                            Map<String, TableIdent> indicesToIdents) {
         this.collectPhaseId = collectPhaseId;
-        this.fetchSize = fetchSize == 0 ? Paging.PAGE_SIZE : Math.min(fetchSize, Paging.PAGE_SIZE);
+        this.fetchSize = calculateFetchSize(
```

Comment:
wouldn't it make sense take the limit into account and only calculate this for bigger resultsets? or even move that calculation out of the constructor at all?

if we for some reason need to serialize this class this would be required anyways.